### PR TITLE
cmd/snapd-apparmor: disable on WSL-1 and 2, fix tests

### PR DESCRIPTION
**Pull request by zyga**
This fixes broken unit tests that don't work on WSL-2 environment due to incorrect assumption about the host environment and also change the non-test code to *disable* attempts to use apparmor on WSL-2.

The reason for this is articulated in 35e15ecdc27a0700108c4c77dbfb7a076b5f5c78 but mainly, because it just doesn't work unless you have a custom kernel and run only one distribution at a time.

Once AppArmor is enabled in the Microsoft distribution of the Linux kernel, and once stacking is enabled in the Microsoft WSL init process, then we can drop most of the special-casing for WSL and just treat WSL-2-with-apparmor-and-apparmor-stacking as a LXD container with internal policy. Before that time comes, this is a more sane approach.

We can consider a follow-up that adds an environment variable like `SNAPD_FORCE_WSL_APPARMOR=1` so that people who do use custom kernels and run only one distribution at a single time, can still use it.

### CODE REVIEW

File: cmd/snapd-apparmor/main.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ import (
 // be properly handled by this function.
 func isContainerWithInternalPolicy() bool {
 	if release.OnWSL {
-		return true
+		return false
```
Comment (olivercalder) [inline cmd/snapd-apparmor/main.go:72]: If this is just a workaround for now, perhaps a comment stating this should be changed once Microsoft fixes things would be good

**Review by olivercalder — COMMENTED** [review]
Looks good, thanks!

File: cmd/snapd-apparmor/main.go (hunk 1 above)
Comment (zyga) [reply cmd/snapd-apparmor/main.go:72]: It's more of an "implements" things but you are right. I will push a comment.

**Review by alexmurray — APPROVED** [review]
LGTM

File: cmd/snapd-apparmor/main.go (hunk 1 above)
Comment (zyga) [reply cmd/snapd-apparmor/main.go:80]: I've now included and pushed a comment.

**Review by olivercalder — APPROVED** [review]
Thanks for the comment, LGTM!

**Pull request merged**